.github/workflows/test_on_push.yaml

```diff
@@ -0,0 +1,39 @@
+name: Run tests
```

Comment:
General thoughs (not a review in fact, I just briefly glanced at Ci changes. I plan to do thorough review on the next week):

1. Please, extract CI enhancements into its own PR. It'll simplify our communication around changes. If you need them to ease work on the code: okay, just base your code / testing changes on your CI changes. We'll push CI changes first, rebase code / testing changes upward and push them.
2. I see that you removed RPM / Deb packages building and deployment. If you don't plan to move it to GitHub Actions, please, keep it on Travis CI (with the idea to move in a future).

Reply:
This PR is small, so it seems to me you can leave this PR with the addition of ci for luatest and one test from it.
I left building and deployment in travis. And new Luatests are launched on gh ci.

Reply:
'This is small' does not advocate mixing of irrelevant changes into one commit. It has no any sense except saving 10 minutes of `git reset HEAD^ && git add -p` for you. If you don't want to pay this time to present you code in a good shape, why should I pay my time to review it?

Reply:
Just a little weird to add ci to testing without a single test, but okay: https://github.com/tarantool/expirationd/pull/55

Reply:
moved it to another PR #55